### PR TITLE
Code improvements / cleanup

### PR DESCRIPTION
**Pull request by offa**
Code improvements and cleanup, mostly to fix some warnings.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] ~~Link to relevant issues in GitHub or Jira~~
- [x] ~~Link to relevant pull requests, esp. upstream and downstream changes~~
- [x] ~~Ensure you have provided tests - that demonstrates feature works or fixes the issue~~ --> Refactoring

### CODE REVIEW

**Review by basil — COMMENTED** [review]
As of jenkinsci/jenkins#4604, the Jenkins project prefers `edu.umd.cs.findbugs.annotations.NonNull` to `javax.annotation.Nonnull`.

**Comment by offa** [comment]
Ah thanks, I'll replace it.

**Comment by offa** [comment]
@basil annotation replaced.

**Review by basil — APPROVED** [review]
> **Update:** @basil Just to be sure before I replace them: Is `NonNull` correct? It's marked as [deprecated](http://findbugs.sourceforge.net/api/edu/umd/cs/findbugs/annotations/NonNull.html)!?

I had the same question a few months back. Apparently they were un-deprecated in recent versions of SpotBugs, so that deprecation warning should eventually disappear once a newer Jenkins core JAR (one with a recent version of SpotBugs) is on the classpath. While  the JSR 305 annotations are unmaintained (despite a lack of deprecation) and have apparently caused actual problems (as described in the Jenkins core PR I linked to earlier).

The changes look fine to me overall. The moving around of lots of unrelated imports seems to add unnecessary noise to the PR, but if it's fine with the maintainers of this repository then it's fine with me. 🤷‍♂️

File: src/main/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleExtension.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package org.jvnet.hudson.test.junit.jupiter;
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleExtension.java:1]: Now a merge conflict I am afraid.

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java [hunk 2]
```diff
@@ -252,6 +183,79 @@
 import org.springframework.dao.DataAccessException;
 import org.xml.sax.SAXException;
 
+import javax.servlet.ServletContext;
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/JenkinsRule.java:186]: Prefer strictly alphabetized `import`s in one block. Not for humans to read anyway.

File: src/main/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleExtension.java (hunk 1 above)
Comment (offa) [reply src/main/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleExtension.java]: Done, merged with master.

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java (hunk 2 above)
Comment (offa) [reply src/main/java/org/jvnet/hudson/test/JenkinsRule.java]: Ok, I have updated the files changed by this PR.

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java [hunk 3]
```diff
@@ -1440,10 +1441,7 @@ public boolean isGoodHttpStatus(int status) {
         if ((400 <= status) && (status <= 417)) {
             return false;
         }
-        if ((500 <= status) && (status <= 505)) {
-            return false;
-        }
-        return true;
+        return (500 > status) || (status > 505);
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/JenkinsRule.java:1444]: Reduces legibility in this case (parallel with previous clauses destroyed), but whatever.

File: src/main/java/org/jvnet/hudson/test/SecuredMockFolder.java [hunk 4]
```diff
@@ -69,15 +69,14 @@ public boolean hasPermission(Permission p) {
...
     
-    private boolean hasPermissionInField(String sid, @Nonnull Permission p) {
+    private boolean hasPermissionInField(String sid, @NonNull Permission p) {
         if (sid.equals(grantedUser)) {
-            if (grantedPermissions != null && grantedPermissions.contains(p.getId())) {
-                return true;
-            }
+            return grantedPermissions != null && grantedPermissions.contains(p.getId());
         }
         return false;
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/SecuredMockFolder.java:76]: Had to read this a couple times to make sure it preserved behavior! Clearer still would be a single

```java
return sid.equals(grantedUser) && grantedPermissions != null && grantedPermissions.contains(p.getId());
```

**Review by jglick — APPROVED** [review]

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java (hunk 3 above)
Comment (offa) [reply src/main/java/org/jvnet/hudson/test/JenkinsRule.java]: Both reverted to if statements.

File: src/main/java/org/jvnet/hudson/test/SecuredMockFolder.java (hunk 4 above)
Comment (offa) [reply src/main/java/org/jvnet/hudson/test/SecuredMockFolder.java]: Changed to single line.

**Pull request merged**